src/cwe_checker_lib/src/analysis/expression_propagation/mod.rs

```diff
@@ -0,0 +1,232 @@
+                });
+
+                Some(insertable_expressions)
+            }
+            Def::Load {
+                var,
+                address: _expression,
+            } => {
+                // Expressions dependent on the assigned variable are no longer insertable
+                insertable_expressions.retain(|input_var, input_expr| {
+                    input_var != var && !input_expr.input_vars().into_iter().any(|x| x == var)
+                });
+                Some(insertable_expressions)
+            }
+            Def::Store { .. } => Some(insertable_expressions),
+        }
+    }
+
+    fn update_call_stub(
+        &self,
+        value_after_call: &Self::Value,
+        _call: &Term<Jmp>,
+    ) -> Option<Self::Value> {
+        Some(value_after_call.clone())
+    }
+
+    fn update_jump(
+        &self,
+        value: &Self::Value,
+        _jump: &Term<Jmp>,
+        _untaken_conditional: Option<&Term<Jmp>>,
+        _target: &Term<Blk>,
+    ) -> Option<Self::Value> {
+        Some(value.clone())
+    }
+
+    fn update_call(
+        &self,
+        _value: &Self::Value,
+        _call: &Term<Jmp>,
+        _target: &Node,
+        _calling_convention: &Option<String>,
+    ) -> Option<Self::Value> {
+        None // This propagation is intra-functional
+    }
+
+    fn update_return(
+        &self,
+        _value: Option<&Self::Value>,
+        _value_before_call: Option<&Self::Value>,
+        _call_term: &Term<Jmp>,
+        _return_term: &Term<Jmp>,
+        _calling_convention: &Option<String>,
+    ) -> Option<Self::Value> {
+        Some(HashMap::new()) // Start with no prior knowledge
+    }
+
+    fn specialize_conditional(
+        &self,
+        value: &Self::Value,
+        _condition: &Expression,
+        _block_before_condition: &Term<Blk>,
+        _is_true: bool,
+    ) -> Option<Self::Value> {
+        Some(value.clone())
+    }
+}
+
+/// Performs the fixpoint algorithm and returns the computation.
+///
+/// Panics, if the computation does not stabilizes.
+fn compute_expression_propagation<'a>(
+    project: &'a Project,
+    graph: &'a Graph,
+) -> Computation<GeneralizedContext<'a, Context<'a>>> {
+    let context = Context::new(project, graph);
+    let mut computation = create_computation(context, None);
+
+    for node in graph.node_indices() {
+        if let Node::BlkStart(_blk, _sub) = graph[node] {
+            // A start in the CFG has no incoming edges in the CFG and
+            // are mostly due to cases where the control flow graph is incomplete.
+            // We assume that no expressions are insertable at such starting nodes.
+            if graph
+                .neighbors_directed(node, petgraph::Incoming)
+                .next()
+                .is_none()
+            {
+                computation.set_node_value(node, NodeValue::Value(HashMap::new()));
+            }
+        }
+    }
+
+    computation.compute_with_max_steps(100);
```

Comment:
We also need to set the `BlkStart` values of blocks that are entry points into functions! Else we will miss most basic blocks in the fixpoint computation.